tests/unittests/test_request_timeout.py

```diff
@@ -0,0 +1,150 @@
+        """
+        mocked_request.side_effect = ConnectionError
+
+        config = {
+            "access_token": "dummy_at",
+            "user_agent": "dummy_ua"
+        }
+        # initialize 'RechargeClient'
+        try:
+            with RechargeClient(
+                config['access_token'],
+                config['user_agent'],
+                config.get('request_timeout')) as client:
+                pass
+        except ConnectionError:
+            pass
+        # verify that we backoff for 5 times
+        self.assertEquals(mocked_request.call_count, 5)
+
+class MockResponse():
+    '''
+    Mock response  object for the requests call 
+    '''
+    def __init__(self, resp, status_code, content=[""], headers=None, raise_error=False, text={}, links=""):
+        self.json_data = resp
+        self.status_code = status_code
+        self.content = content
+        self.headers = headers
+        self.raise_error = raise_error
+        self.text = text
+        self.reason = "error"
+        self.links = links
+
+    def prepare(self):
+        return (self.json_data, self.status_code, self.content, self.headers, self.raise_error)
+
+    def json(self):
+        return self.text
+
+class TestRequestTimeoutValue(unittest.TestCase):
+    '''
+    Test that request timeout parameter works properly in various cases
+    '''
+    @mock.patch('tap_recharge.client.requests.Session.request', return_value = MockResponse("", status_code=200))
+    @mock.patch('tap_recharge.client.RechargeClient.check_access_token')
+    def test_config_provided_request_timeout(self, mock_get, mock_request):
+        """ 
+            Unit tests to ensure that request timeout is set based on config value
+        """
+        config = {"access_token": "dummy_at", "user_agent": "dummy_ua", "request_timeout": 100}
+        client = RechargeClient(**config)
+        client.request("GET", "dummy_path")
+        
+        mock_request.assert_called_with('GET', 'https://api.rechargeapps.com/dummy_path', stream=True, timeout=100.0, headers={'X-Recharge-Access-Token': 'dummy_at', 'Accept': 'application/json', 'User-Agent': 'dummy_ua'})
+
+    @mock.patch('tap_recharge.client.requests.Session.request', return_value = MockResponse("", status_code=200))
+    @mock.patch('tap_recharge.client.RechargeClient.check_access_token')
+    def test_default_value_request_timeout(self, mock_get, mock_request):
+        """ 
+            Unit tests to ensure that request timeout is set based default value
+        """
+        config = {"access_token": "dummy_at", "user_agent": "dummy_ua"}
+        client = RechargeClient(**config)
+        client.request("GET", "dummy_path")
+        
+        mock_request.assert_called_with('GET', 'https://api.rechargeapps.com/dummy_path', stream=True, timeout=300.0, headers={'X-Recharge-Access-Token': 'dummy_at', 'Accept': 'application/json', 'User-Agent': 'dummy_ua'})
```

Comment:
@namrata270998 Wouldn't the value of timeout be "300"?
Can you please check the code logic for the same?

Reply:
Right, it would be 300. Updated it to 300